src/transxchange/naptan.rs

```diff
@@ -107,13 +125,25 @@ where
         .map(|record: csv::Result<NaPTANStopInArea>| {
             record.with_context(|_| "Error parsing the CSV record into a StopInArea")
         })
+        .filter(|record| {
```

Comment:
I'm not fan of this. It's really difficult to understand, it seems tricky.
a for loop would increase the understanding imho

@ArnaudOggy  feel free to merge

Reply:
I extracted a function instead of transforming into a `for` loop. It's not perfect but it helps the readability.